app/controllers/telegram_webhooks_controller.rb

```diff
@@ -0,0 +1,119 @@
+class TelegramWebhooksController < Telegram::Bot::UpdatesController
+  include Telegram::Bot::UpdatesController::MessageContext
+  include Telegram::Bot::UpdatesController::CallbackQueryContext
+
+  def start!(*)
+    if session?
+      respond_with :message, text: 'Choose action', reply_markup: menu_markup
+    else
+      save_context :auth
+      respond_with :message, text: 'Hi! Give me your mobile number.', reply_markup: phone_markup
+    end
+  end
+
+  def auth
+    user = User.find_by(mobile: update['message']['contact']['phone_number'])
+    if user
+      session[:user_id] = user.id
+      session[:session_key] = session_key
+    else
+      respond_with(:message, text: "Not found user.")
+    end
+    start!
+  end
+
+  def add_meter_reading!(*)
+    if session[:session_key]
+      save_context :add_meter
+      respond_with :message, text: "Choose Utility provider", reply_markup: {
+        keyboard: user.companies.map { |c| [{ text: c.name }] }, resize_keyboard: true, one_time_keyboard: true
+      }
+    else
+      start!
+    end
+  end
+
+  def add_meter(*company_name)
+    session[:company] = Company.find_by(name: company_name.join(' ')).id
+    save_context :save_meter_reading
+    respond_with :message, text: 'Write Meter Reading'
+  end
+
+  def save_meter_reading(value, *)
+    billing_contract = user.billing_contracts.find_by(company_id: session[:company])
+
+    meter_reading = billing_contract.meter_readings.build(value: to_number(value))
+
+    if meter_reading.save
+      respond_with :message, text: "Saved Meter Reading #{value}.", reply_markup: menu_markup
+    else
+      respond_with :message, text: "Can't add meter reading #{value}.", reply_markup: menu_markup
+    end
+  end
+
+  def companies!(*)
+    session? ? respond_with(:message, text: user.companies.map(&:name).join(', ')) : start!
```

Comment:
return respond_with(:message, ...) if session?
start!

Reply:
also you can move this user.companies.map(&:name).join(', ')
to something like user.companies_for_output